spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/cassandra/CassandraMetricsBinder.java

```diff
@@ -0,0 +1,153 @@
+		TIMER_SNAPSHOT_METRICS.put("p99", Snapshot::get99thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("p98", Snapshot::get98thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("p95", Snapshot::get95thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("p75", Snapshot::get75thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("min", Snapshot::getMin);
+		TIMER_SNAPSHOT_METRICS.put("max", Snapshot::getMax);
+		TIMER_SNAPSHOT_METRICS.put("mean", Snapshot::getMean);
+		TIMER_SNAPSHOT_METRICS.put("median", Snapshot::getMedian);
+		TIMER_SNAPSHOT_METRICS.put("stdDev", Snapshot::getStdDev);
+	}
+
+	static {
+		TIMER_METRICS.put("mean_rate", Timer::getMeanRate);
+		TIMER_METRICS.put("m1_rate", Timer::getOneMinuteRate);
+		TIMER_METRICS.put("m5_rate", Timer::getFiveMinuteRate);
+		TIMER_METRICS.put("m15_rate", Timer::getFifteenMinuteRate);
+	}
+
+	private final Iterable<Tag> tags;
+
+	private final CqlSession session;
+
+	private CassandraMetricsSettings cassandraMetricsSettings;
+
+	public CassandraMetricsBinder(@NonNull CqlSession cqlSession,
+			@NonNull CassandraMetricsSettings cassandraMetricsSettings) {
+		this.cassandraMetricsSettings = cassandraMetricsSettings;
+		this.session = cqlSession;
+		this.tags = Tags.concat(Collections.emptyList(), "sessionName", cqlSession.getName());
+	}
+
+	@Override
+	public void bindTo(@NonNull MeterRegistry meterRegistry) {
+		Assert.isTrue(this.session.getMetrics().isPresent(), "Metrics on the CqlSession needs to be present");
+
+		this.session.getMetrics().get().getRegistry().getMetrics().forEach((name, m) -> {
+			if (m instanceof Timer) {
+				registerTimer(name, (Timer) m, meterRegistry);
+			}
+			else if (m instanceof Counting) {
+				registerMeter(name, (Counting) m, meterRegistry);
+			}
+			else if (m instanceof com.codahale.metrics.Gauge) {
+				registerGauge(name, (com.codahale.metrics.Gauge) m, meterRegistry);
+			}
+		});
+	}
+
+	private void registerTimer(String metricName, Timer timer, MeterRegistry meterRegistry) {
+		String name = String.format("%s.%s", SESSION_PREFIX, metricName);
+
+		// count metric should be registered at the root level of the metric name
+		FunctionCounter.builder(name, timer, Timer::getCount).tags(this.tags).description(DESCRIPTION)
+				.register(meterRegistry);
+
+		TIMER_METRICS.forEach((n, supplier) -> {
+			String nameTimerMetric = String.format("%s.%s", name, n);
+			FunctionCounter.builder(nameTimerMetric, timer, (t) -> supplier.apply(timer).doubleValue()).tags(this.tags)
+					.baseUnit(TimeUnit.SECONDS.name()) // all rates are calculated per
+					// second
+					.description(DESCRIPTION).register(meterRegistry);
+		});
+
+		TIMER_SNAPSHOT_METRICS.forEach((n, supplier) -> {
+			String nameTimerMetric = String.format("%s.%s", name, n);
+			FunctionCounter
+					.builder(nameTimerMetric, timer,
+							(t) -> convertToTimeUnit(supplier.apply(timer.getSnapshot()).doubleValue()))
+					.tags(this.tags).baseUnit(this.cassandraMetricsSettings.getTimerTimeUnit().name())
```

Comment:
I've removed this conversion, and the CassandraMetricsSettings class in general.

Reply:
Why do you want to remove it? Don't you think that converting may be useful for end-users? 
(I don't know if spring allows that for other modules)

Reply:
Honestly nanoseconds is the best you can do.

Reply:
The reason for the conversion was to allow users to control the granularity of the timer snapshot. Nanoseconds is the finest we can go, but maybe that is a bit overkill for some situations (thinking about web apps in general here).

Initially, we had another attribute in the settings class that would allow you to disable metrics, even if some of them were enabled in the driver config. This is not necessary if we are contributing metrics to Spring Boot as we can disable them via the Spring metrics management settings (built-in). So that leaves just this one config attribute in CassandraMetricsSettings.

So the question is, is having a configurable time unit for these metrics worth the extra class and setting? If so, I can bring it back, but I will have to resume searching for what documentation needs to go where for the setting as one of the gradle documentation tasks (`documentConfigurationProperties`) fails for all the attributes in CassandraMetricsSettings.

Reply:
I think it will be easier to just go with nanosecond granularity here. I don't know that there is a lot of value in trying to give users the flexibility to control it.